src/3p-frame-messaging.js

```diff
@@ -125,6 +125,38 @@ export function isAmpMessage(message) {
       message.indexOf('{') != -1);
 }
 
+
+/**
+ * Returns true if win's properties can be accessed and win is defined.
+ * This functioned is used to determine if a window is cross-domained
+ * from the perspective of the current window.
+ * @param {Window} win
+ * @return {boolean}
+ */
+export function canInspectWindow(win) {
```

Comment:
IMO, only centralize code if and when there are more than two consumers but your call.  If you do decide to leave here, you need explicit test coverage.

Reply:
moved